examples/uploader/_localrepo.py

```diff
@@ -0,0 +1,133 @@
+        self.key_dir = key_dir
+        if not os.path.isdir(self.key_dir):
+            os.makedirs(self.key_dir)
+
+        self.base_url = base_url
+
+        self.updater = Updater(
+            metadata_dir=metadata_dir,
+            metadata_base_url=f"{base_url}/metadata/",
+        )
+        self.updater.refresh()
+
+    @property
+    def targets_infos(self) -> Dict[str, MetaFile]:
+        raise NotImplementedError  # we never call snapshot
+
+    @property
+    def snapshot_info(self) -> MetaFile:
+        raise NotImplementedError  # we never call timestamp
+
+    def open(self, role: str) -> Metadata:
+        """Return cached (or fetched) metadata"""
+
+        # if there is a metadata version fetched from remote, use that
+        # HACK: access Updater internals
+        # pylint: disable=protected-access
+        if role in self.updater._trusted_set:
+            return copy.deepcopy(self.updater._trusted_set[role])
+
+        # otherwise we're creating metadata from scratch
+        md = Metadata(Targets())
+        # this makes version bumping in close() simpler
+        md.signed.version = 0
+        return md
+
+    def close(self, role: str, md: Metadata) -> None:
+        """Store a version of metadata. Handle version bumps, expiry, signing"""
+        md.signed.version += 1
+        md.signed.expires = datetime.utcnow() + self.expiry_period
+
+        with open(f"{self.key_dir}/{role}", "rt", encoding="utf-8") as f:
+            signer = SSlibSigner(json.loads(f.read()))
+
+        md.sign(signer, append=False)
+
+        # Upload using "api/role"
+        uri = f"{self.base_url}/api/role/{role}"
+        r = requests.post(uri, data=md.to_bytes(JSONSerializer()), timeout=5)
+        r.raise_for_status()
+
+    def add_target(self, role: str, targetpath: str) -> bool:
+        """Add target to roles metadata and submit new metadata version"""
+
+        # HACK: make sure we have the roles metadata in updater._trusted_set
+        # (or that we're publishing the first version)
+        try:
+            self.updater.get_targetinfo(targetpath)
+        except RepositoryError:
+            # HACK Assume this is because we're just publishing version 1
+            # (so the roles metadata does not exist on server yet)
+            pass
+
+        data = bytes(targetpath, "utf-8")
+        targetfile = TargetFile.from_data(targetpath, data)
+        try:
+            with self.edit(role) as delegated:
+                delegated.targets[targetpath] = targetfile
+
+        except Exception as e:  # pylint: disable=broad-except
+            print(f"Failed to submit new {role} with added target: {e}")
+            return False
+
+        print(f"Uploaded role {role} v{delegated.version}")
+        return True
+
+    def add_delegation(self, role: str) -> bool:
+        """Use the (unauthenticated) delegation adding API endpoint"""
+        keydict = keys.generate_ed25519_key()
```

Comment:
When reading the docs, I was wondering, where the signing key came from. It makes sense to create it here, and it might also make sense to not go into details in the docs. Just want to mention that it wasn't obvious to me.

Reply:
Never mind, I see now that the `uploader` cli help message includes this bit of information.

Reply:
I agree it's not obvious, added a mention in README as well